nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -431,7 +435,7 @@ in
       };
 
       services.zfs.zed.settings = {
-        ZED_EMAIL_PROG = mkDefault "${pkgs.mailutils}/bin/mail";
+        ZED_EMAIL_PROG = mkIf cfgZED.enableMail (mkDefault "${pkgs.mailutils}/bin/mail");
```

Comment:
This also has to somehow override the `zfs` package with `enableMail`.

Reply:
maybe we can add enableMail to `passthru` and just reference that.

Reply:
Done in 01f6aee23cf6f464e2b300bf274fc31ce408d65f and b87934f6cf97d2a2eec2aa7c7fe26eb87061629d.

Reply:
Wait, I forgot to override zfs (lol), another commit incoming.

Reply:
Should I go the assertion approach, or the override approach? I feel they're mutually exclusive.